code/modules/spells/ability_types/realized.dm

```diff
@@ -789,3 +789,88 @@
 	var/mob/living/carbon/human/H = owner
 	H.adjust_attribute_buff(JUSTICE_ATTRIBUTE, -30)
 	H.remove_status_effect(STATUS_EFFECT_LCBURN)
+
+/* Yang - Duality */
+/obj/effect/proc_holder/ability/tranquility
+	name = "Tranquility"
+	desc = "An ability that does massive white damage in an area and heals people's health and sanity. \
+	Healing someone that's wearing harmony of duality will grant them huge buffs to their defenses and stats."
+	action_icon_state = "yangform0"
+	base_icon_state = "yangform"
+	cooldown_time = 60 SECONDS
+
+	var/damage_amount = 300 // Amount of explosion damage
+	var/explosion_range = 15
+
+/obj/effect/proc_holder/ability/tranquility/Perform(target, mob/user)
+	cooldown = world.time + (1.5 SECONDS)
+	if(!do_after(user, 1 SECONDS))
+		to_chat(user, "<span class='warning'>You must stand still to explode!</span>")
+		return
+	new /obj/effect/temp_visual/explosion/fast(get_turf(user))
+	var/turf/orgin = get_turf(user)
+	var/list/all_turfs = RANGE_TURFS(explosion_range, orgin)
+	var/mob/living/carbon/human/H = user
+	for(var/i = 0 to explosion_range)
+		for(var/turf/T in all_turfs)
+			if(get_dist(user, T) > i)
+				continue
+			new /obj/effect/temp_visual/dir_setting/speedbike_trail(T)
+			for(var/mob/living/L in T)
+				if(H.faction_check_mob(L, FALSE))
+					continue
+				if(L.stat == DEAD)
+					continue
+				L.apply_damage(damage_amount, WHITE_DAMAGE, null, L.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
+			for(var/mob/living/carbon/human/L in T)
+				if(!H.faction_check_mob(L, FALSE))
+					continue
+				if(L.status_flags & GODMODE)
+					continue
+				if(L.stat == DEAD)
+					continue
+				if(L.is_working) //no work heal :(
+					continue
+				L.adjustBruteLoss(-120)
+				L.adjustSanityLoss(-120)
+				var/obj/item/clothing/suit/armor/ego_gear/realization/duality_yin/Z = L.get_item_by_slot(ITEM_SLOT_OCLOTHING)
+				if(istype(Z))
+					H.apply_status_effect(/datum/status_effect/duality_yang)
+					new /obj/effect/temp_visual/healing(get_turf(L))
+			all_turfs -= T
```

Comment:
```suggestion
	for(var/i = 0 to explosion_range)
		for(var/turf/T in all_turfs)
			if(get_dist(user, T) > i)
				continue
			new /obj/effect/temp_visual/dir_setting/speedbike_trail(T)
			user.HurtInTurf(damage_amount, list(), WHITE_DAMAGE)
			for(var/mob/living/carbon/human/L in T)
				if(!user.faction_check_mob(L, FALSE))
					continue
				if(L.stat == DEAD)
					continue
				if(L.is_working) //no work heal :(
					continue
				L.adjustBruteLoss(-120)
				L.adjustSanityLoss(-120)
				if(istype(L.get_item_by_slot(ITEM_SLOT_OCLOTHING), /obj/item/clothing/suit/armor/ego_gear/realization/duality_yin))
					H.apply_status_effect(/datum/status_effect/duality_yang)
					new /obj/effect/temp_visual/healing(get_turf(L))
			all_turfs -= T
```
I will also note that the healing VFX only applies to those who have the opposing suit on at the moment.